diff --git a/sys/msdk/gstmsdkdec.c b/sys/msdk/gstmsdkdec.c
index 51e7057b9..55e1ae45c 100644
--- a/sys/msdk/gstmsdkdec.c
+++ b/sys/msdk/gstmsdkdec.c
@@ -1065,6 +1065,7 @@ gst_msdkdec_handle_frame (GstVideoDecoder * decoder, GstVideoCodecFrame * frame)
 
       if (bitstream.DataLength == 0) {
         flow = GST_FLOW_OK;
+        surface = NULL;
         break;
       }
     } else if (status == MFX_ERR_MORE_DATA) {
